booking/src/main/java/org/dcsa/conformance/standards/booking/checks/BookingChecks.java

```diff
@@ -0,0 +1,645 @@
+        issues.add("Port of Discharge/Place of Delivery value must be provided");
+      }
+      if (podNode == null || podNode.isEmpty()) {
+        issues.add("Port of Load/Place of Receipt values must be provided");
+      }
+      if(!"SD".equals(receiptTypeAtOrigin) && ielNode != null) {
+        issues.add("Container intermediate export stop-off location should not be provided");
+      }
+
+      return issues;
+    });
+
+
+  private static final JsonContentCheck REQUESTED_CHANGES_PRESENCE = JsonAttribute.customValidator(
+    "Requested changes must be present for the selected Booking Status ",
+    (body) -> {
+      var bookingStatus = body.path("bookingStatus").asText("");
+      var issues = new LinkedHashSet<String>();
+      if (PENDING_CHANGES_STATES.contains(BookingState.fromWireName(bookingStatus))) {
+        var requestedChanges = body.get("requestedChanges");
+        if (requestedChanges == null) {
+          issues.add("requestedChanges is missing in allowed booking states %s".formatted(PENDING_CHANGES_STATES));
+        }
+      }
+      return issues;
+    }
+  );
+
+  private static final JsonContentCheck REASON_PRESENCE = JsonAttribute.customValidator(
+    "Reason field must be present for the selected Booking Status",
+    (body) -> {
+      var bookingStatus = body.path("bookingStatus");
+      var amendedBookingStatus = body.path("amendedBookingStatus");
+      var issues = new LinkedHashSet<String>();
+      var status = amendedBookingStatus.isMissingNode() || amendedBookingStatus.isNull() ? bookingStatus : amendedBookingStatus;
+      if (REASON_STATES.contains(BookingState.fromWireName(status.asText()))) {
+        var reason = body.get("reason");
+        if (reason == null) {
+          issues.add("reason is missing in the Booking States %s".formatted(REASON_STATES));
+        }
+      }
+      return issues;
+    }
+  );
+
+  private static final JsonContentCheck CHECK_CONFIRMED_BOOKING_FIELDS = JsonAttribute.customValidator(
+    "check confirmed booking fields availability",
+    body -> {
+      var issues = new LinkedHashSet<String>();
+      var bookingStatus = body.path("bookingStatus").asText("");
+      if (CONFIRMED_BOOKING_STATES.contains(BookingState.fromWireName(bookingStatus))) {
+        if (body.get("confirmedEquipments") == null) {
+          issues.add("confirmedEquipments for confirmed booking is not present");
+        }
+        if (body.get("termsAndConditions") == null) {
+          issues.add("termsAndConditions for confirmed booking is not present");
+        }
+        if (body.get("transportPlan") == null) {
+          issues.add("transportPlan for confirmed booking is not present");
+        }
+        if (body.get("carrierClauses") == null) {
+          issues.add("carrierClauses for confirmed booking is not present");
+        }
+        if (body.get("charges") == null) {
+          issues.add("charges for confirmed booking is not present");
+        }
```

Comment:
This is not correct - it is correct that `charges` can only be part of a Confirmed booking - but if there are no charges then it is ok to not provide the field (e.g. the field is not required)